pkg/commands/update/versioner.go

```diff
@@ -101,91 +81,83 @@ func (g GitHub) LatestVersion(ctx context.Context) (semver.Version, error) {
 	}
 	defer rc.Close()
 
-	var extension string
-
-	// TODO: We might need to also account for Window users by also checking for
-	// the .zip extension that goreleaser generates:
-	// https://github.com/fastly/cli/blob/26588cfd2d00d18643bac5cc18242b2d5ee84b34/.goreleaser.yml#L51
-	//
-	// Ideally the formats would be the same, but if that's not possible then
-	// we can look to use a genericised method such as
-	// https://pkg.go.dev/github.com/mholt/archiver#Extract for handling the
-	// extraction of a binary from the asset file instead of using the current
-	// archiver.NewTarGz().Extract() method.
-	if strings.HasSuffix(g.releaseAsset, ".tar.gz") {
-		extension = ".tar.gz"
+	dir, err := os.MkdirTemp("", "fastly-download")
+	if err != nil {
+		return "", fmt.Errorf("error creating release download directory: %w", err)
 	}
+	defer os.RemoveAll(dir)
 
-	tmp := os.TempDir()
-	dst, err := os.CreateTemp(tmp, fmt.Sprintf("%s_%s_*%s", g.binary, version, extension))
+	dst, err := os.Create(filepath.Join(dir, g.releaseAsset))
 	if err != nil {
-		return filename, fmt.Errorf("error creating temp release asset file: %w", err)
+		return "", fmt.Errorf("error creating release asset file: %w", err)
 	}
 
 	_, err = io.Copy(dst, rc)
 	if err != nil {
-		return filename, fmt.Errorf("error downloading release asset: %w", err)
+		return "", fmt.Errorf("error downloading release asset: %w", err)
 	}
 
 	if err := dst.Close(); err != nil {
-		return filename, fmt.Errorf("error closing release asset file: %w", err)
+		return "", fmt.Errorf("error closing release asset file: %w", err)
 	}
 
-	assetFile := dst.Name()
+	tempFile := dst.Name()
 
-	// If we need to perform an extraction, create a temporary directory and do it there,
-	// then copy the extracted file back to the temporary one so we just track one file.
+	// TODO: We might need to also account for Window users by also checking for
+	// the .zip extension that goreleaser generates:
+	// https://github.com/fastly/cli/blob/26588cfd2d00d18643bac5cc18242b2d5ee84b34/.goreleaser.yml#L51
+	//
+	// Ideally the formats would be the same, but if that's not possible then
+	// we can look to use a genericised method such as
+	// https://pkg.go.dev/github.com/mholt/archiver#Extract for handling the
+	// extraction of a binary from the asset file instead of using the current
+	// archiver.NewTarGz().Extract() method.
 	if strings.HasSuffix(g.releaseAsset, ".tar.gz") {
-		dir, err := os.MkdirTemp(tmp, "extract")
-		if err != nil {
-			return filename, fmt.Errorf("error creating temp extraction directory: %w", err)
-		}
-		defer os.RemoveAll(dir)
-		if err := archiver.NewTarGz().Extract(assetFile, g.binary, dir); err != nil {
-			return filename, fmt.Errorf("error extracting binary: %w", err)
-		}
-		if err := os.Rename(filepath.Join(dir, g.binary), assetFile); err != nil {
-			return filename, fmt.Errorf("error renaming binary: %w", err)
+		if err := archiver.NewTarGz().Extract(tempFile, g.binary, dir); err != nil {
```

Comment:
So then this would become

```suggestion
		if err := archiver.NewTarGz().Extract(assetFile.Name(), g.binary, dir); err != nil {
```